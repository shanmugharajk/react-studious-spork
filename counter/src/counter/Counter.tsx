import "./counter.css";
import { observer } from "mobx-react-lite";

import type { ICounterStore } from "./counter-store";

interface IProps {
  counterStore: ICounterStore;
}

export const Counter: React.FunctionComponent<IProps> = observer(
  ({ counterStore }) => (
    <div className="Counter__container">
      <span data-testid="counter-lbl" className="Counter__clicks-lbl">
        Clicks {counterStore.count}
      </span>

      <div className="Counter__footer-container">
        <button
          className="Counter__btn"
          onClick={() => counterStore.increment()}
        >
          Increment
        </button>
        <button
          className="Counter__btn"
          onClick={() => counterStore.decrement()}
        >
          Decrement
        </button>
      </div>
    </div>
  )
);
