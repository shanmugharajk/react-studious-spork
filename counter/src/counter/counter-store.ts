import { makeAutoObservable } from "mobx";

export interface ICounterStore {
  count: number;
  increment: () => void;
  decrement: () => void;
}

export class CounterStore implements ICounterStore {
  count = 0;

  constructor() {
    makeAutoObservable(this);
  }

  increment() {
    this.count += 1;
  }

  decrement() {
    this.count -= 1;
  }
}

export const counterStore = new CounterStore();
