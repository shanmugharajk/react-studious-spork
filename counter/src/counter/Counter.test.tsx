import { fireEvent, render, screen } from "@testing-library/react";

import { Counter } from "./Counter";
import { CounterStore } from "./counter-store";

describe("Counter", () => {
  const counterStore = new CounterStore();

  it("should render Counter component", () => {
    const { container } = render(<Counter counterStore={counterStore} />);

    expect(container).toMatchSnapshot();
  });

  it("should increment and decrement count properly", () => {
    render(<Counter counterStore={counterStore} />);

    expect(screen.getByTestId("counter-lbl")).toHaveTextContent("Clicks 0");

    fireEvent.click(screen.getByText("Increment"));
    expect(screen.getByTestId("counter-lbl")).toHaveTextContent("Clicks 1");

    fireEvent.click(screen.getByText("Decrement"));
    expect(screen.getByTestId("counter-lbl")).toHaveTextContent("Clicks 0");
  });
});
