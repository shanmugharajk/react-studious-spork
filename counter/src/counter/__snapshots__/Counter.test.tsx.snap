// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render Counter component 1`] = `
<div>
  <div
    class="Counter__container"
  >
    <span
      class="Counter__clicks-lbl"
      data-testid="counter-lbl"
    >
      Clicks 
      0
    </span>
    <div
      class="Counter__footer-container"
    >
      <button
        class="Counter__btn"
      >
        Increment
      </button>
      <button
        class="Counter__btn"
      >
        Decrement
      </button>
    </div>
  </div>
</div>
`;
